quickwit/quickwit-ui/src/components/QueryEditor/AggregationEditor.tsx

```diff
@@ -0,0 +1,305 @@
+    if (pos >= agg.length) {
+      options.push((
+          <MenuItem value="new">Add aggregation</MenuItem>
+      ))
+    }
+    let addHistogram = true;
+    let addTerm = true;
+    for(let i = 0; i < agg.length; i++) {
+      if (i == pos) continue;
+      if (getAggregationKind(agg[i]) === "histogram") addHistogram = false;
+      if (getAggregationKind(agg[i]) === "term") addTerm = false;
+    }
+    if (addHistogram) {
+      options.push((<MenuItem value="histogram">Histogram aggregation</MenuItem>))
+    }
+    if (addTerm) {
+      options.push((<MenuItem value="term">Term aggregation</MenuItem>));
+    }
+    if (agg.length > 1) {
+      options.push((
+          <MenuItem value="rm">Remove aggregation</MenuItem>
+      ))
+    }
+    return options;
+  }
+
+  // do the initial filling of parameters
+  const aggregationConfig = props.searchRequest.aggregationConfig;
+  if (aggregationConfig.histogram === null && aggregationConfig.term === null) {
+    const initialAggregation = Object.assign({}, ...aggregations);
+    const initialSearchRequest = Object.assign({}, props.searchRequest, {aggregationConfig: initialAggregation});
+    props.onSearchRequestUpdate(initialSearchRequest);
+  }
+
+  const drawAdditional = (pos: number, aggs: ({term: TermAgg} | {histogram: HistogramAgg})[]) => {
+    const agg = aggs[pos]
+    if (isHistogram(agg)) {
+      return (
+        <Select
+          value={agg.histogram.interval}
+          onChange={(e) => handleHistogramChange(pos, e)}
+          sx={{ "margin-left": "10px", "min-height": "44px" }}
+        >
+          <MenuItem value="auto">Auto</MenuItem>
+          <MenuItem value="10s">10 secods</MenuItem>
+          <MenuItem value="1m">1 minute</MenuItem>
+          <MenuItem value="5m">5 minutes</MenuItem>
+          <MenuItem value="10m">10 minutes</MenuItem>
+          <MenuItem value="1h">1 hour</MenuItem>
+          <MenuItem value="1d">1 day</MenuItem>
+        </Select>
+      );
+    }
+    if (isTerm(agg)) {
+      return [(
+          <TextField 
+            variant="standard"
+            label="Field"
+            onChange={(e) => handleTermFieldChange(pos, e)}
+            sx={{ "margin-left": "10px" }}
+          />
+        ),(
+          <TextField 
+            variant="standard"
+            label="Return top"
+            type="number"
+            onChange={(e) => handleTermCountChange(pos, e)}
+            value={agg.term.size}
+            sx={{ "margin-left": "10px" }}
+          />
+        )]
+    }
+    return (null);
+  }
+
+  return (
+    <Box>
+      <FormControl variant="standard" sx={{ m: 1, minWidth: 120, display: 'flex', flexDirection: 'row', }}>
+        <Select
+          value={getAggregationKind(aggregations[0])}
+          onChange={(e) => handleAggregationChange(0, e)}
+          sx={{ "min-height": "44px", width: "190px" }}
+        >
+          { makeOptions(0, aggregations) }
+        </Select>
+        {drawAdditional(0, aggregations)}
```

Comment:
FormControl is meant to hold only one input component. Multiple inputs will throw errors in console.